web-marketplace/src/pages/BuyCredits/BuyCredits.Form.tsx

```diff
@@ -465,6 +554,47 @@ export const BuyCreditsForm = ({
         modalState={modalState}
         redirectRoute={`${projectHref.replace(/^\//, '')}/buy`}
       />
+      {warningModalContent.current && (
+        <BuyWarningModal
+          modalContent={warningModalContent.current.modalContent}
+          warningModalState={warningModalState}
+          onClose={setWarningModalState}
+          handleClick={action => {
+            if (action === KEPLR_LOGIN_REQUIRED) {
+              if (!activeWalletAddr) {
+                // no connected wallet address
+                setConnectWalletModal(atom => void (atom.open = true));
+              } else {
```

Comment:
you're missing a condition here, it should be:

```suggestion
              } else if (!isConnected) {
```

check out `setupWalletModal` prop in BuyCredits.Form.tsx

then we need a final `else` (web3 account connected), where we should just switch to crypto payment option and go back to step 1

Reply:
Can you elaborate why we need this final `else`? is this to handle cases where users have logged in using WalletConnect?

Reply:
it could but not necessarily, it's just in the case where we've got a web3 user connected (to the right account in Keplr too), in this case, if the user clicks on "or buy with crypto", then we just want to select the crypto payment option on the first step

Reply:
You also need to close the modal "Sorry, another user has purchased all of the USD credits", now it does go back to the first step but the modal should close automatically before:

![image](https://github.com/user-attachments/assets/aab523d1-83d4-4c3f-a39e-9f6e13423e7f)